tests/Behat/Services/Factory/FailedRefundPaymentFactory.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tests\Sylius\RefundPlugin\Behat\Services\Factory;
+
+use Sylius\RefundPlugin\Entity\RefundPaymentInterface;
+use Sylius\RefundPlugin\Factory\RefundPaymentFactoryInterface;
+
+final class FailedRefundPaymentFactory implements RefundPaymentFactoryInterface
+{
+    private const FAILED_FILE = __DIR__.'/refund-payment-failed.json';
+
+    /** @var RefundPaymentFactoryInterface */
+    private $baseRefundPaymentFactory;
+
+    public function __construct(RefundPaymentFactoryInterface $baseRefundPaymentFactory)
+    {
+        $this->baseRefundPaymentFactory = $baseRefundPaymentFactory;
+    }
+
+    public function createWithData(
+        string $orderNumber,
+        int $amount,
+        string $currencyCode,
+        string $state,
+        int $paymentMethodId
+    ): RefundPaymentInterface {
+        if (file_exists(self::FAILED_FILE)) {
```

Comment:
I guess we should delete that file on `__construct` to make sure that we're always starting from the same point. Eg. if the app breaks between `failRefundPaymentCreation` and `createWithData`, we're left in an incorrect state.

Reply:
Apparently, it does not work as expected :/ The constructor is called twice and removes the file unnecessarily. I've added a `@AfterScenario` hook to the context, to do the trick 🖖